File: rd-cpp/src/rd_framework_cpp/src/test/util/TestScheduler.cpp
//
// Created by jetbrains on 03.10.2018.
//

#include "TestScheduler.h"

namespace rd {
	namespace test {
		namespace util {
			void TestScheduler::queue(std::function<void()> action) {
				action();
			}

			bool TestScheduler::is_active() const {
				return true;
			}
		}
	}
}

File: rd-kt/rd-text/src/test/kotlin/com/jetbrains/rd/rdtext/test/util/TestScheduler.kt
package com.jetbrains.rd.rdtext.test.util

import com.jetbrains.rd.util.reactive.ExecutionOrder
import com.jetbrains.rd.util.reactive.IScheduler

object TestScheduler : IScheduler {
    override fun flush() {}
    override fun queue(action: () -> Unit) = action()
    override val isActive: Boolean get() = true
    override val executionOrder: ExecutionOrder
        get() = ExecutionOrder.Unknown
}